### PR TITLE
[Docs] Added NSIS install steps to build-windows.md

### LINKED ISSUE #14699: docs: Windows build docs are missing `make deploy`
When building for Windows (using `doc/build-windows.md`), I discovered the following steps were necessary but undocumented, in order to generate the installer:
```
sudo apt install nsis
make deploy
makensis share/setup.nsi
```

### PR DESCRIPTION
**Pull request by LucianaMarques**
This pull request is intended to close #14699 

I never built for Windows, so I'm not sure if the instructions I added are correct, please give me feedback on it. 

Thank you!

### CODE REVIEW

File: doc/build-windows.md [hunk 1]
```diff
@@ -67,6 +67,17 @@ build process.
 
 See also: [dependencies.md](dependencies.md).
 
+## Install NSIS
+
+The following steps are necessary to create the binary file. In order to create it, the system must
```
Comment (fanquake) [inline doc/build-windows.md]: s/binary file/binary installer

**Comment by fanquake** [comment]
I'd suggest moving this down between the Depends and Installation sections of the build doc. Someone is not going to be able to run `make deploy` unless that have already completed the steps for a 32/64 bit build.

**Review by ken2812221 — CHANGES_REQUESTED** [review]

File: doc/build-windows.md (hunk 1 above)
Comment (LucianaMarques) [reply doc/build-windows.md]: I didn't understand your comment. Should I replace 'binary file' to 'binary installer', or the commands in the terminal should run in 's/binary file/binary installer/' directory?

**Comment by LucianaMarques** [comment]
> I'd suggest moving this down between the Depends and Installation sections of the build doc. Someone is not going to be able to run `make deploy` unless that have already completed the steps for a 32/64 bit build.

Done!

File: doc/build-windows.md (hunk 1 above)
Comment (fanquake) [reply doc/build-windows.md]: You should replace binary file with binary installer

File: doc/build-windows.md [hunk 2]
```diff
@@ -129,6 +129,17 @@ Then build using:
 
 For further documentation on the depends system see [README.md](../depends/README.md) in the depends directory.
 
+## Install NSIS
```
Comment (fanquake) [inline doc/build-windows.md]: Can you change the title to be something like "Create Installer", and use the dash title method i.e ------ below the words (the same as all the rest of the titles in this document).

Comment (laanwj) [reply doc/build-windows.md]: well the `depends system` above it also uses `##`, it's inconsistent with the rest of the document that's true

Comment (LucianaMarques) [reply doc/build-windows.md]: I guess I see @fanquake 's logic. Done.

**Comment by LucianaMarques** [comment]
> The last step (calling `makensis` manually) should not be necessary. `configure.ac` has logic to find `makensis`, you'll need to install it _before_ calling `./configure` for that:
> 
> https://github.com/bitcoin/bitcoin/blob/master/configure.ac#L470
> 
> Then, `make deploy` will generate the installer for you:
> 
> https://github.com/bitcoin/bitcoin/blob/master/Makefile.am#L73
> 
> (see, for example, how the gitian build script works: https://github.com/bitcoin/bitcoin/blob/master/contrib/gitian-descriptors/gitian-win.yml#L157)

Thank you very much for the explanation!

File: doc/build-windows.md [hunk 3]
```diff
@@ -129,6 +129,17 @@ Then build using:
 
 For further documentation on the depends system see [README.md](../depends/README.md) in the depends directory.
 
+Create installer
+-----------------
+
+The following steps are necessary to create the binary installer. In order to create it, the system must
+have [NSIS](https://nsis.sourceforge.io/Main_Page) installed.
```
Comment (laanwj) [inline doc/build-windows.md]: I think you should mention that it should be installed *before* running `./configure`, if you install it just before the `make` step in the steps below it's not going to work.

**Comment by fanquake** [comment]
@LucianaMarques Thanks for your work here. I've picked this up in #14950. The changes are still attributed to you.

**Pull request closed without merging**